src/coreclr/vm/threadsuspend.cpp

```diff
@@ -2933,6 +2933,18 @@ BOOL Thread::RedirectThreadAtHandledJITCase(PFN_REDIRECTTARGET pTgt)
     if (!IsContextSafeToRedirect(pCtx))
         return (FALSE);
 
+    // In some rare cases the stack pointer may be outside the stack limits.
+    // SetThreadContext would fail assuming that we are trying to bypass CFG.
+    // 
+    // NB: the check here is slightly more strict than what OS requires,
```

Comment:
Can we suppress the assert below for this situation instead of upfront check?

Reply:
Right, makes sense, since this is a rare case.

Reply:
Hmm. OS actually does more complicated check - it would allow SP to be within a guard page past the limit, if limit is not the real limit (as in `teb->DeallocationStack`).

Reply:
Actually that is irrelevant. Using this check to suppress the assert is still an equivalent of this PR change, since the check is more strict.

Reply:
I will make the change.